src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -863,7 +863,13 @@ private void DoCloseHelper()
                 }
 
                 if ((hostRunspace == null) || (this == hostRunspace))
-                {
+                {                   
+                    // We should close transcripting only if we are running in default runspace.
+                    if (executionContext.CurrentRunspace != null && DefaultRunspace != null && executionContext.CurrentRunspace.Id != DefaultRunspace.Id)
```

Comment:
This is tricky because DefaultRunspace is a thread static property and so there can be multiple DefaultRunspaces for multiple threads.  It is really hard to know when all runspaces are closed.  We do now keep a list of all (non-disposed) local runspaces.  See Connection.cs https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/hostifaces/Connection.cs#L799
You could search through this list and if there is only one runspace left in the "open" state (meaning this is the last runspace) then do the stop transcribing.
This list only shows runspaces that have not yet been disposed.  But a runspace can be non-open and still be in the list if it was not disposed, so you need to check the state of each runspace.